client/store.js

```diff
@@ -0,0 +1,64 @@
+import { createStore, applyMiddleware } from 'redux'
+import thunkMiddleware from 'redux-thunk'
+import { composeWithDevTools } from 'redux-devtools-extension'
+import axios from 'axios'
+
+//ACTION TYPES
+const SET_USER = 'SET_USER'
+const SET_PRODUCTS = 'SET_PRODUCTS'
+
+//ACTION CREATORS
+const setUser = user => ({
+    type: SET_USER,
+    user
+})
+
+const setProducts = products => ({
+    type: SET_PRODUCTS,
+    products
+})
+
+//THUNK CREATORS
+export const fetchProducts = () => async dispatch => {
+    try {
+        const response = await axios.get('api/products')
+        const products = response.data
+        return dispatch(setProducts(products))
+    } catch (error) { console.log(error) } //or do whatever else we want with the error
```

Comment:
Prof suggests that if you `catch` the error in the thunk, then you should `throw` it and then deal with the error when you are calling the thunk.
```suggestion
    } catch (error) { throw new Error(error) } 
```